packages/monaco/src/browser/monaco-editor-provider.ts

```diff
@@ -184,4 +191,13 @@ export class MonacoEditorProvider {
         };
     }
 
+    createDiffNavigator(editor: TextEditor): DiffNavigator {
+        if (editor instanceof MonacoDiffEditor) {
+            const options = this.createOptions(this.diffPreferencePrefixes);
```

Comment:
The diff navigator won't react on preference changes?

Reply:
they should, right?

Reply:
After looking at the implementation of `DiffNavigator`: It seems that we will need to create a new instance and sync it with the current to react to changes properly. It is complicated, let's wait till someone requests it.

Reply:
actually, I'm think, if we need it to be initialized together with the diff editor, we can/might/should keep it a part of `MonacoDiffEditor` and we should have a provider for diff navigator instead of a factory. I cannot see how it would make sense to have two different navigators for same diff editor.